imap_processing/tests/mag/test_mag_decom.py

```diff
@@ -1,72 +1,54 @@
-import dataclasses
 from pathlib import Path
 
-from imap_processing.mag.l0.decom_mag import decom_packets
+import pandas as pd
 
+from imap_processing.cdf import global_attrs
+from imap_processing.mag.l0.decom_mag import decom_packets, export_to_xarray
 
-def test_burst_decom():
-    current_directory = Path(__file__).parent
-    burst_test_file = current_directory / "mag_single_burst.pkts"
-    l0 = decom_packets(str(burst_test_file))[0]
 
-    assert l0.ccsds_header.PKT_APID == 1068
-    assert l0.ccsds_header.SRC_SEQ_CTR == 1
-    assert l0.ccsds_header.PKT_LEN == 1821
+def test_mag_decom():
+    current_directory = Path(__file__).parent
+    burst_test_file = current_directory / "mag_l0_test_data.pkts"
+    l0 = decom_packets(str(burst_test_file))
 
-    expected = dataclasses.replace(
-        l0,
-        SHCOARSE=435954632,
-        PUS_STYPE=21,
-        PUS_SSUBTYPE=3,
-        COMPRESSION=0,
-        MAGO_ACT=1,
-        MAGI_ACT=1,
-        PRI_SENS=0,
-        PRI_VECSEC=6,
-        SEC_VECSEC=3,
-        PRI_COARSETM=435954625,
-        PRI_FNTM=64610,
-        SEC_COARSETM=435954625,
-        SEC_FNTM=64607,
-    )
-    assert l0 == expected
+    expected_output = pd.read_csv(current_directory / "mag_l0_test_output.csv")
 
+    index = 0
+    for test in l0:
```

Comment:
```suggestion
    for index, test in enumerate(l0):
```
and remove the index += at the end of the loop